server/src/main/java/org/opensearch/index/IndexService.java

```diff
@@ -1584,4 +1609,60 @@ public boolean clearCaches(boolean queryCache, boolean fieldDataCache, String...
         return clearedAtLeastOne;
     }
 
+    /**
+     * This method is called when the refresh level is changed from index level to shard level or vice versa.
+     */
+    public void onRefreshLevelChange(boolean newShardLevelRefreshVal) {
+        synchronized (refreshMutex) {
+            if (this.shardLevelRefreshEnabled != newShardLevelRefreshVal) {
+                boolean prevShardLevelRefreshVal = this.shardLevelRefreshEnabled;
+                this.shardLevelRefreshEnabled = newShardLevelRefreshVal;
+                // The refresh mode has changed from index level to shard level
```

Comment:
nit: It can be other way round as well, right?

Reply:
Yes, updated the comment